### ISSUE TITLE
Unicode and non-printable codes in array

### ISSUE THREAD

**Issue by Ketan-Desai** [cemdervis/SharpConfig#56]
Hi,

I want char[] as follows, how do I put ini file?

`char[] trimChars = { ' ', (char)160, '-', '_', '\'', '\"', '<', '>', '\n', '\r', '\t', ':', '?', (char)30, '\0', '=' };`

Thanks

**Comment by Ketan-Desai** [comment]
i.e. unicode support will be required (to enclose in string array)?

**Comment by Ketan-Desai** [comment]
or a feature like Char, CharArray canbe implemented?

**Comment by Ketan-Desai** [comment]
also, why not always trim double quotes from StringValue and StringValueArray?

**Comment by Ketan-Desai** [comment]
[Setting]
DateSuffix = { " 00:00:00", " 12:00:00 AM", " 0:00:00", " 0:00" }
TimePrefix = "12/31/1899 "
CsvSeparator1 = ",;|~^"
CsvSeparator2 = {9}

Leading and Trailing spaces are getting trimmed (DateSuffix, TimeSuffix)

Non-printable codes can't be put, hence required int[] and later cast to char[] work-arround required.
(CsvSeparator1 & 2)

StringValueTrimmed must remove quote char only! and required same functionality for StringValueArray

**Comment by Ketan-Desai** [comment]
TrimChar = " -_<>:?"

StringValueTrimmed throws exception

**Comment by Ketan-Desai** [comment]
There is some compatibility issue in v3.1.0 & v3.2.0 related to StringValueArray.

The leading and trailing spaces, including double quotes are getting trimmed.

**Comment by cemdervis** [comment]
Please post example code that produces the wrong results.

**Comment by Ketan-Desai** [comment]
```
            var cfg = Configuration.LoadFromFile(@"D:\Sample.ini");

            var array = cfg["Data"]["Values"].StringValueArray;

            for (int i = 0; i < array.Length; i++)
            {
                Console.WriteLine(string.Format("[{0}]", array[i]));
            }
```

Sample.ini

[Data]
Values = {" One "," Two "}

Output using 3.1.0
[" One "]
[" Two "]

Output using v3.2.0
[One]
[Two]

In v3.2.0 Quotes and Spaces are trimmed.

Thanks

**Comment by cemdervis** [comment]
I'm going to look into it this weekend, thanks.

**Comment by cemdervis** [comment]
The second issue is fixed and will be released soon. What exactly was the issue in your first comment? Please open separate issues for multiple issues.

**Comment by cemdervis** [comment]
Fixed with version 3.2.1.
